opentelemetry_collector/receiver/dockerstats/scraper.go

```diff
@@ -41,15 +41,37 @@ var (
 		Type:        mpb.MetricDescriptor_GAUGE_INT64,
 		LabelKeys:   []*mpb.LabelKey{containerNameLabel},
 	}
+	// Container health metrics.
+	uptimeDesc = &mpb.MetricDescriptor{
+		Name:        "container/uptime",
+		Description: "Container uptime",
+		Unit:        "s",
+		Type:        mpb.MetricDescriptor_GAUGE_INT64,
```

Comment:
This and some of your other metrics seem like the sort of thing you would want to view as a distribution. As a gauge, this is only really useful if it is filtered to the level of an individual VM and container. While as a distribution the metric could easily be used to view the distribution of container uptimes across all the VMs in a project or all the VMs in a region for example. If you want a distribution metricspb.MetricDescriptor_GAUGE_DISTRIBUTION is the correct value.

Reply:
In the context of a single Flex instance (where these metrics are defined), none of them are a distribution. For example, uptime of a single instance can't be viewed as a distribution, it would always be an exact value. Your feedback about reshaping some of these metrics into distributions as we aggregate them is valid. The aggregations would likely be exported as separate metrics.

Reply:
You can view a metric like this filtered to a single instance as a distribution. It would just be a distribution with a single bucket containing a value per time step. This is also how the vm image age metric is handled, as a distribution which only contains one value per VM instance. I'd prefer using a distribution since a) you can still get the single instance information from a distribution b) generally it's a bad idea to design a metric so it's only meaningful when filtered to certain labels. (This is listed under "Worst Practices" on go/streamz-guide.)

Reply:
Point b) is a mischaracterization of this metric. If you look at Metrics Explorer in Google Cloud Console today, you have various aggregations available including percentiles. So you can still easily see what the 50th and 95th percentile uptime for containers in your app look like when you have multiple instances. I think that's what you were alluding to when you wanted a distribution?

On a single instance level though, uptime only has one value at a given point in time. Compare that to request latency, for example, which is a common example of a distribution metric. The reason it is created as a distribution is that because it has multiple values. And recording 50th, 95th and 99th percentile on each instance, then aggregating them across all instances to get 50th, 95th and 99th percentile for the service yields less accurate results than a distribution of latencies across buckets (assuming correctly chosen bucket size and count). If you map it back to uptime metric now, there is no 50th, 95th and 99th percentile uptime for a single instance. There is just... uptime. I think the same logic applies to VM image age and it shouldn't be a distribution either. That won't stop you from seeing the distribution of VM image age across the fleet.

Happy to chat more over IM / GVC to resolve this.